zkvm/src/predicate.rs

```diff
@@ -68,18 +69,16 @@ impl Predicate {
         encoding::write_point(&self.to_point(), prog);
     }
 
-    /// Verifies whether the current predicate is a disjunction of two others.
+    /// Verifies whether the current predicate is a disjunction of n others.
     /// Returns a `PointOp` instance that can be verified in a batch with other operations.
-    pub fn prove_or(&self, left: &Predicate, right: &Predicate) -> PointOp {
-        let l = left.to_point();
-        let r = right.to_point();
-        let f = Self::commit_or(l, r);
+    pub fn prove_disjunction(&self, preds: Vec<Predicate>) -> PointOp {
```

Comment:
let's pass `pred: &[Predicate]`, so we don't have to move in the vec. Then we don't need to clone the `k`-th predicate it in `VM::select`.